vscode/microsoft-kiota/.vscode/settings.json

```diff
@@ -12,5 +12,11 @@
     "typescript.tsc.autoDetect": "off",
     "cSpell.words": [
         "Kiota"
-    ]
+    ],
+    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
```

Comment:
will all contributors have access to this extension? I think this should be in your user settings. Otherwise add this extension to recommendations. I think a better way to enforce formatting is on CI

Reply:
Yes, linting on the build pipeline shall work, doesn't it?
Please do not introduce dependencies on other extensions.

Reply:
The problem with pipeline re-formatting the code is what whoever saved the code, (via commit & push), will have slightly different code on the server, so even if they are the only one who uploaded code, they have to do a pull.

Also, a pipeline failure because of formatting is also not something I'd want to enforce because it would just make the developer experience suffer. It is fairly easy to set up a "format on save" on the editor. Having it in a shared editor configuration just helps make it consistent with everyone

Reply:
The problems you outlined on the developer experience will apply to all developers that don't use VSCode though. At least with CI, we can make sure a contributor using a different text editor will not check in unformatted code.

Reply:
I see. It would make sense in a less specific case like ours. The configs are in the context of .vscode a folder available to users using vscode to build vscode extensions.

Also we might be getting lost here because AFAIK, we are debating whether it's worth it for vscode contributors using vscode to 'formatOnSave' instead of making formatting depend on a pipeline run